src/SDK/MonitoringApplianceEmulator/Controls/HypertextBlock.cs

```diff
@@ -0,0 +1,135 @@
+        /// This method replace all link patterns within the new assigned hypertext  <see cref="Hyperlink"/> elements.
+        /// </summary>
+        /// <param name="dependencyObject">The dependency object</param>
+        /// <param name="eventArgs">The event args</param>
+        private static void OnHypertextPropertyChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs eventArgs)
+        {
+            if (!(dependencyObject is HypertextBlock hypertextTextBlock))
+            {
+                throw new ArgumentException($"The dependency object must be of type {typeof(HypertextBlock)}, but it is from type {dependencyObject.GetType()}.", nameof(dependencyObject));
+            }
+
+            if (!(eventArgs.NewValue is string hypertext))
+            {
+                throw new ArgumentException($"The new value must be of type {typeof(string)}, but it is from type {eventArgs.NewValue.GetType()}.", nameof(eventArgs));
+            }
+
+            Match match = HtmlHyperlinkRegex.Match(hypertext);
+
+            // In case there are no links in the text, just add it
+            if (!match.Success)
+            {
+                hypertextTextBlock.Inlines.Add(hypertext);
+            }
+
+            // Go over all link pattern matches and replace it with a hyperlink element
+            Match previousMatch = null;
+            while (match.Success)
+            {
+                // First, add text before the current match
+                int startIndexOfTextBeforeMatch = previousMatch == null ?
+                    0 :
+                    previousMatch.Index + previousMatch.Length;
+
+                int lengthOfTextBeforeMatch = previousMatch == null ?
+                    match.Index :
+                    match.Index - (previousMatch.Index + previousMatch.Length);
+
+                string textBeforeMatch = hypertext.Substring(startIndexOfTextBeforeMatch, lengthOfTextBeforeMatch);
+
+                hypertextTextBlock.Inlines.Add(textBeforeMatch);
+
+                // Second, convert the match to hyperlink and add it
+                string url = match.Groups["href"].Value.Trim('\'', '\"');
+                string linkDisplayText = match.Groups["linkText"].Value;
+
+                Hyperlink link = new Hyperlink { IsEnabled = true };
+                link.Inlines.Add(linkDisplayText);
+                link.NavigateUri = new Uri(url);
+                link.RequestNavigate += (sender, args) => Process.Start(args.Uri.ToString());
+
+                hypertextTextBlock.Inlines.Add(link);
+
+                // Continue to next match
+                previousMatch = match;
+                match = match.NextMatch();
+            }
+
+            // If there is text after the last found match, add it as well
+            if (previousMatch != null)
+            {
+                int remainingTextStartIndex = previousMatch.Index + previousMatch.Length;
```

Comment:
>remainingTextStartIndex [](start = 20, length = 23)

check that there is anything to add - better not just add an empty inline if possible